client/src/Component/EngineerCard/index.jsx

```diff
@@ -1,21 +1,33 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import SingleCard from './SingleCard';
 import './style.css';
 
 function EngCard() {
+  const [info, setInfo] = useState([]);
+  useEffect(() => {
+    const abortController = new AbortController();
+    fetch('/api/engHome', {
+      signal: abortController.signal,
+    })
+      .then((res) => res.json())
+      .then((data) => setInfo(data.data))
+      .catch((err) => console.log(err));
+    return () => abortController.abort();
```

Comment:
don't use `fetch` anymore use `axios` instead of it
and use the `async` & `await` syntax

Reply:
you can search how to make a `cancelToken` in axios
instead of `AbortController` for clean up